src/signed_data.rs

```diff
@@ -243,36 +245,95 @@ impl<'a> SubjectPublicKeyInfo<'a> {
     }
 }
 
-/// A signature algorithm.
-pub struct SignatureAlgorithm {
+/// A detail-less error when a signature is not valid.
+#[derive(Debug, Copy, Clone)]
+pub struct InvalidSignature;
+
+/// An abstract signature algorithm, usable for verifying signatures.
+pub trait SignatureAlgorithm: Send + Sync {
+    /// Return the `AlgorithmIdentifier` that must be present on a `subjectPublicKeyInfo`
+    /// for this `SignatureAlgorithm` to be considered.
```

Comment:
editorial nit: maybe "    /// for this `SignatureAlgorithm` to be considered for verification" ?